### ISSUE TITLE
Unnecessary comparison false positive with union of Coroutine and None

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#7881]
**Describe the bug**

It seems the `reportUnnecessaryComparison` check is overwritten if `Coroutine` is part of a union.

```python
from typing import Any, Coroutine

def func() -> Coroutine[Any, Any, int] | None: ...

coro = func()
if coro:  # error
    pass
```

```
test.py:6:4 - error: Conditional expression references coroutine which always evaluates to True (reportUnnecessaryComparison)
```

**VS Code extension or command-line**
pyright 1.1.362

**Config**
```json
"reportUnnecessaryComparison": "error"
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.363, which I just published.
